exchanges/account/account.go

```diff
@@ -292,6 +302,85 @@ func (s *Service) Save(incoming *Holdings, creds *Credentials) error {
 	return errs
 }
 
+// Update updates the balance for a specific exchange and credentials
+func (s *Service) Update(exch string, changes []Change, creds *Credentials) error {
+	if exch == "" {
+		return fmt.Errorf("cannot update balance: %w", errExchangeNameUnset)
+	}
+
+	if creds.IsEmpty() {
+		return fmt.Errorf("cannot update balance: %w", errCredentialsAreNil)
+	}
+
+	exch = strings.ToLower(exch)
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	accounts, ok := s.exchangeAccounts[exch]
+	if !ok {
+		id, err := s.mux.GetID()
+		if err != nil {
+			return err
+		}
+		accounts = &Accounts{
+			ID:          id,
+			subAccounts: make(map[Credentials]map[key.SubAccountAsset]currencyBalances),
+		}
+		s.exchangeAccounts[exch] = accounts
+	}
+
+	var subAccounts map[key.SubAccountAsset]currencyBalances
+	subAccounts, ok = accounts.subAccounts[*creds]
+	if !ok {
+		subAccounts = make(map[key.SubAccountAsset]currencyBalances)
+		accounts.subAccounts[*creds] = subAccounts
+	}
+
+	var errs error
+	for _, change := range changes {
+		if !change.AssetType.IsValid() {
+			errs = common.AppendError(errs, fmt.Errorf("cannot update balance for %s [%s] %w",
+				change.Account, change.AssetType, asset.ErrNotSupported))
+			continue
+		}
+		if change.Balance == nil {
+			errs = common.AppendError(errs, fmt.Errorf("cannot update balance for %s [%s] %w",
+				change.Account, change.AssetType, errBalanceIsNil))
+			continue
+		}
+
+		accAsset := key.SubAccountAsset{
+			SubAccount: change.Account,
+			Asset:      change.AssetType,
+		}
+		assets, ok := subAccounts[accAsset]
+		if !ok {
+			assets = make(map[*currency.Item]*ProtectedBalance)
+			accounts.subAccounts[*creds][accAsset] = assets
+		}
+		if change.Balance != nil {
```

Comment:
With the `change.Balance != nil` check above, this check can be dropped like a risky asset after a global tariff